File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.java
import java.util.*;

class Solution
{
    public int solution(int [][]board)
    {
        double answer = 0;
        int[][] dp = new int[board.length+1][board[0].length + 1];
        
        for (int y = 0; y < board.length; y++) {
            for (int x = 0; x < board[0].length; x++) {
                dp[y+1][x+1] = board[y][x] == 0 ? 0 : 1 + (int) Math.min(Math.min(dp[y][x], dp[y+1][x]), dp[y][x+1]);
                answer = Math.max(answer, dp[y+1][x+1]);
            }
        }
        
        return (int) (answer * answer);
    }
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    
    for j in range(1, len(board)):
        for i in range(1, len(board[0])):
            if board[j][i] != 0:
                board[j][i] += min(board[j-1][i], board[j][i-1], board[j-1][i-1])
    
    answer = 0
    
    for b in board:
        answer = max(answer, max(b))
    

    return answer ** 2